src/App.vue

```diff
@@ -603,13 +604,26 @@ export default class App extends Mixins(CommonMixin, DateMixin, FilingTemplateMi
     await updateLdUser(key, email, firstName, lastName, custom)
   }
 
+  /** Perform high level component validations before proceeding to summary. */
+  private async validateComponents (): Promise<void> {
+    // Verify component validations
+    await this.setValidResolutionDate(this.getIsResolutionDatesValid)
+
+    // evaluate valid flags. Scroll to invalid components or continue to review.
+    if (this.validateAndScroll(this.getValidComponentFlags, ComponentFlags)) {
+      // We don't change views just interchange components, so scroll to top for better UX.
+      this.scrollToTop(document.getElementById('app'))
```

Comment:
I just had a look at `scrollToTop`. It's actually asynchronous, which makes sense because we the scroll transition takes some ticks to complete. However, it's not properly written to await.

Could you please fix `scrollToTop` and then await it everywhere it's used?

(ie, we want the scroll to complete before we set summary mode to true)

Reply:
For sure, will do!

Reply:
Updated in next commit!